sound/soc/sof/intel/hda.h

```diff
@@ -435,6 +434,11 @@ struct sof_intel_hda_stream {
 	(SOF_HDA_ADSP_SD_ENTRY_SIZE * ((s)->index) \
 	 + SOF_HDA_ADSP_LOADER_BASE)
 
+struct sof_stream_ctx {
+	struct hdac_stream *hstream;
+	size_t posn_offset;
+};
+
```

Comment:
I don't think we use "ctx" postfix in any structs now, maybe best to leave it out.

And actually, why not just use "struct sof_intel_hda_stream" to store whatever HDA specific data we need to access?
I.e. not add new structs, but put the data to the existing struct.

Then in code you can get from private_data to hstream by casting, and with hstream_to_sof_hda_stream() you get the full struct. Or maybe I'm missing something. :) But this would seem to be the simplest option.

Reply:
> I don't think we use "ctx" postfix in any structs now, maybe best to leave it out.

@kv2019i ctx is removed and adding some missing code. 

> 
> And actually, why not just use "struct sof_intel_hda_stream" to store whatever HDA specific data we need to access?
> I.e. not add new structs, but put the data to the existing struct.

My goal is to get soundwire stream in pcm ops. I am not sure is it a HDA specific data or not.  It seems that we use HDA in front end and soundwire in back end.

> 
> Then in code you can get from private_data to hstream by casting, and with hstream_to_sof_hda_stream() you get the full struct. Or maybe I'm missing something. :) But this would seem to be the simplest option.

Thanks for your comment. The naming of hstream_to_sof_hda_stream() is confusing. We will get the pointer from hdac_ext_stream but not  hdac_stream.  And which in private_data is hdac_stream. We pre-allocate these stuff in [hda_dsp_stream_init](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda-stream.c#L597). But soundwire stream will be allocated at pcm_open. So it may not be easy to use the same way in soundwire stream.

Reply:
@bardliao Hmm, but hda_dsp_stream_init() is done already at probe, so the allocations are already there.
So if you put hdac_ext_stream as private data, then the above works.

I think a separate "sof_stream" is still misleading. This is a struct only used with Intel HDA hardware, so the "sof_" prefix is not correct. And when you try to think of other names, you realize we already have "sof_intel_hda_stream" and the two have overlapping purposes. I'd still try to see if we can just use one struct for this purpose.

I think it's ok to add soundwire handle to "sof_intel_hda_stream". Soundwire is one of the outputs from the Intel HDA module (others are HDA-codec, SSP/I2S), so it's ok to use this same struct I think.

Reply:
> @bardliao Hmm, but hda_dsp_stream_init() is done already at probe, so the allocations are already there.
> So if you put hdac_ext_stream as private data, then the above works.
> 
> I think a separate "sof_stream" is still misleading. This is a struct only used with Intel HDA hardware, so the "sof_" prefix is not correct. And when you try to think of other names, you realize we already have "sof_intel_hda_stream" and the two have overlapping purposes. I'd still try to see if we can just use one struct for this purpose.
> 
> I think it's ok to add soundwire handle to "sof_intel_hda_stream". Soundwire is one of the outputs from the Intel HDA module (others are HDA-codec, SSP/I2S), so it's ok to use this same struct I think.

@kv2019i Thanks for the explanation and I agree with you. Let's move all hda related data to sof_intel_hda_stream. I will close this PR and send another PR for that when soundwire is ready. Thanks.